addon/addon/template-cache.js

```diff
@@ -0,0 +1,8 @@
+/**
+ * Originally: https://github.com/emberjs/ember.js/blob/28444d536fa20debef2a67e2f18c5eb11113a4b5/packages/%40ember/-internals/glimmer/lib/template_registry.ts#L9
+ *
+ * import { TEMPLATES } from 'ember';
+ *
+ * Removed for RFC 1003
+ */
+export let TEMPLATES = {};
```

Comment:
Is this a breaking change? If people could previously add to Ember.TEMPLATES?

Reply:
it's a private undocumented API, so if their code breaks, that's on them

Reply:
https://github.com/ember-learn/deprecation-app/pull/1381/files#diff-3fafbe781f6a2ecf159f098078a8a9576a7e7802a50ab4c416de87324974a17dR14

Reply:
another example: https://github.com/emberjs/ember.js/issues/20632#issuecomment-2020478590

🙈

Reply:
It was definitely public once which is probably why I remember it. 

I think it would be reasonable to make this a breaking change for ember-resolver, just so anybody that updates it before ember-source (& the deprecation) knows to look a bit closer. Doesn't really cost anything.

Reply:
> It was definitely public once 

uuggghh gross :(

> I think it would be reasonable to make this a breaking change for ember-resolver,

fair enough! 🎉

Reply:
There's no point in keeping this. Just drop it. 

`Ember.TEMPLATES` is the thing that gets auto-populated from your HTML like:

```html
<script type="text/x-handlebars" data-template-name="my-component">hello world</script>
```

A new major of ember-resolver could document that it no longer supports that pattern.

If people are manually stuffing templates into `Ember.TEMPLATES`, we don't want them to refactor that code to stuff the templates into this template-cache instead. Tell them to use [register](https://api.emberjs.com/ember/5.9/classes/Owner/methods/register?anchor=register).

Reply:
done